src/ast/query.rs

```diff
@@ -577,6 +582,23 @@ impl fmt::Display for Fetch {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum LockType {
+    Shared,
+    Exclusive,
```

Comment:
Since this is always parsed to/from `SHARE` and `FOR UPDATE` what do you think about using `Share` and `Update` as the enum variant names so that the Rust structures more closely mirror the parsed SQL structure and is more consistent with the rest of this crate?

Reply:
I'm not insistent, It's just that `Update` doesn't look like a type of lock.  
If you still think update is better for this crate, I will make changes. Looking forward to reply :->

Reply:
I think for this crate it is better to match the original SQL as closely as possible. 

Downstream crates that actually implement a database can then then map that SQL structure on to actual locking implementations (like shared / exclusive, etc)

Reply:
I got it, thanks for the reply !!